src/services/ecommerce/payments/hooks.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  useQuery,
+  useQueryClient,
+  UseQueryOptions,
+  useMutation,
+} from "@tanstack/react-query";
+import { paymentsApi } from "../client";
+import type {
+  PaymentsApiPaymentsBasketsListRequest,
+  PaymentsApiPaymentsBasketsCreateFromProductCreateRequest,
+  PaymentsApiPaymentsBasketsAddDiscountCreateRequest,
+  PaymentsApiPaymentsCheckoutCreateRequest,
+} from "../generated/v0/api";
+
+const usePaymentsBasketList = (
+  options: PaymentsApiPaymentsBasketsListRequest,
+  opts: Omit<UseQueryOptions, "queryKey"> = {},
```

Comment:
My suggestion here was a little bit bad... it necessitated the typecasting you did elsewhere. What happened is: Because now TS thinks `opts` *might* contain `queryFn`, it can no longer infer the correct shape of the data.

If you do:
```ts
type ExtraQueryOptions = Omit<UseQueryOptions, "queryKey" | "queryFn"> = {}
const usePaymentsBasketList = (
  options: PaymentsApiPaymentsBasketsListRequest,
  opts: ExtraQueryOptions = {
) => {...}
```
all will be well.

Reply:
FWIW - I did this and it was still pretty confused, so the additional casting was left in place.

Reply:
Ah. I guess I've done 
```ts
type ExtraQueryOptions = Pick<UseQueryOptions, "enabled" | ...any_others_we_care>
```
in the past. Some of the more complicated options do see to mess with TS.